source/logging/logging.rst

```diff
@@ -0,0 +1,380 @@
+==============================
+Client Side Operations Timeout
+==============================
+
+:Title: Logging
+:Status: Accepted
+:Minimum Server Version: N/A
+
+.. contents::
+
+--------
+
+Abstract
+========
+This specification defines requirements for drivers' logging configuration and behavior. 
+
+
+META
+====
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”,
+“SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this
+document are to be interpreted as described in `RFC 2119
+<https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+-----
+Terms
+-----
+Structured logging
+    Structured logging refers to producing log messages in a structured format, i.e. a series of
+    key-value pairs, which can be converted to external formats such as JSON.
+
+Unstructured logging
+    Unstructured logging refers to producing string log messages which embed all attached
+    information within that string.
+
+---------------------------
+Implementation requirements
+---------------------------
+Drivers SHOULD implement support for logging in a manner that is idiomatic for their language and
+ecosystem.
+
+
+Minimizing Required Code Changes
+--------------------------------
+If possible, drivers SHOULD support the following configuration requirements in a manner that does
+not require any changes to an application's source code, and in the case of compiled languages,
+does not require recompilation.
+However, if that is impossible to do without conflicting with idiomatic logging patterns for the
```

Comment:
while we had a scope goal around requiring no recompilation but per discussion in the project Slack channel ended up loosening the requirement.

iirc it would be un-idiomatic for .NET to support with no code change at all (@BorisDog could provide more context on this.) 
and it is infeasible for Rust to support without recompilation right now - the only crates available to us for logging/tracing support are pre-1.0 and so it is best practice to gate their usage behind a feature flag (e.g. the Tokio async runtime we depend on does the same thing).